src/megameklab/com/ui/dialog/PrintQueueDialog.java

```diff
@@ -0,0 +1,225 @@
+    private final JList<String> queuedUnitList = new JList<>();
+
+    public PrintQueueDialog(JFrame parent, boolean printToPdf) {
+        super(parent, true, "PrintQueueDialog", "PrintQueueDialog.windowName.text");
+        this.parent = parent;
+        this.printToPdf = printToPdf;
+        initialize();
+    }
+
+    @Override
+    protected Container createCenterPane() {
+        addFromCacheButton.addActionListener(e -> selectFromCache());
+        addFromCacheButton.setMnemonic('a');
+        addFromFileButton.addActionListener(e -> selectFromFile());
+        addFromFileButton.setMnemonic('f');
+        removeButton.addActionListener(e -> removeSelectedUnits());
+        removeButton.setEnabled(false);
+        removeButton.setMnemonic('r');
+        oneUnitPerSheetCheck.setAlignmentX(JComponent.CENTER_ALIGNMENT);
+        oneUnitPerSheetCheck.setToolTipText("When unchecked, the record sheets for some unit types may be printed on the same page. " +
+                "Note that the result may depend on whether reference tables are printed. This can be changed in the Settings.");
+        queuedUnitList.setSelectionMode(ListSelectionModel.MULTIPLE_INTERVAL_SELECTION);
+        queuedUnitList.addListSelectionListener(e -> removeButton.setEnabled(!queuedUnitList.isSelectionEmpty()));
+        queuedUnitList.setVisibleRowCount(15);
+
+        JPanel buttonPanel = new FixedXYPanel(new GridLayout(4, 1));
+        buttonPanel.add(new JLabel());
+        buttonPanel.add(addFromCacheButton);
+        buttonPanel.add(addFromFileButton);
+        buttonPanel.add(removeButton);
+        buttonPanel.setAlignmentY(JComponent.TOP_ALIGNMENT);
+        JScrollPane queuedUnitListScrollPane = new JScrollPane(queuedUnitList);
+        queuedUnitListScrollPane.setAlignmentY(JComponent.TOP_ALIGNMENT);
+        queuedUnitListScrollPane.setBorder(new TitledBorder("Selected Units:"));
+
+        Box centerPanel = Box.createHorizontalBox();
+        centerPanel.add(buttonPanel);
+        centerPanel.add(Box.createHorizontalStrut(30));
+        centerPanel.add(queuedUnitListScrollPane);
+        centerPanel.setBorder(new EmptyBorder(20, 30, 20, 30));
+
+        Box panel = Box.createVerticalBox();
+        panel.add(centerPanel);
+        panel.add(oneUnitPerSheetCheck);
+        panel.add(Box.createVerticalStrut(20));
+        return panel;
+    }
+
+    @Override
+    protected JPanel createButtonPanel() {
+        JPanel panel = new JPanel(new FlowLayout(FlowLayout.RIGHT));
+        var printButton = new MMButton("printButton", resources.getString("PrintQueueDialog.Print.text"),
+                resources.getString("PrintQueueDialog.Print.toolTipText"), this::okButtonActionPerformed);
+        printButton.setMnemonic('p');
+        panel.add(printButton);
+        panel.add(new MMButton("cancelButton", resources.getString("PrintQueueDialog.Cancel.text"),
+                resources.getString("PrintQueueDialog.Cancel.toolTipText"), this::cancelActionPerformed));
```

Comment:
I recommend using the name, resources, *.text, *.toolTipText, action event setup for MMButtons. Shorter while maintaining the resources links.